pkg/vm/emit/emit.go

```diff
@@ -59,11 +59,37 @@ func Int(w *io.BinWriter, i int64) {
 		val := opcode.Opcode(int(opcode.PUSH1) - 1 + int(i))
 		Opcodes(w, val)
 	default:
-		bigInt(w, big.NewInt(i))
+		bigInt(w, big.NewInt(i), true)
 	}
 }
 
-func bigInt(w *io.BinWriter, n *big.Int) {
+// BigInt emits big-integer to the given buffer.
+func BigInt(w *io.BinWriter, n *big.Int) {
+	bigInt(w, n, true)
+}
+
+func bigInt(w *io.BinWriter, n *big.Int, trySmall bool) {
+	if w.Err != nil {
+		return
+	}
+	if trySmall && n.IsInt64() {
+		ni := n.Int64()
+		switch {
```

Comment:
Move this switch into an internal function of its own?

Reply:
Fixed.